lib/nexthop_group.h

```diff
@@ -50,6 +50,8 @@ void copy_nexthops(struct nexthop **tnh, const struct nexthop *nh,
 uint32_t nexthop_group_hash_no_recurse(const struct nexthop_group *nhg);
 uint32_t nexthop_group_hash(const struct nexthop_group *nhg);
 void nexthop_group_mark_duplicates(struct nexthop_group *nhg);
+void nexthop_group_add_sorted(struct nexthop_group *nhg,
```

Comment:
So we put this one in the `_private.h` header to discourage developers not familiar with the intricacies of shared nexthop groups from messing with a groups structure since they are `immutable` now.

If someone not knowing what they are doing calls this, then we could have a situation where the hash of an NHE hash changed and its stuck in limbo.

Reply:
so ... the new structs were all made completely public, and so there's really nothing preventing anyone's code from monkeying with them, right?  the thing that's being added to by the users of this api is not immutable - it's a thing that's being built up by zapi code, or by netlink code, or by zebra_route code internally. there needs to be a way to add nexthops to a list of nexthops - and meet the requirements of the new `hash_entry` semantics - so it seemed worthwhile to just acknowledge what was already being done.

Reply:
That's a good point and maybe we don't need this to be private anymore with this patch since you have to go through the NHE to get to the nexthop_group now anyway.

My main concern before was with a developer doing like a `nexhop_group_add_sorted(re->ng, nh)`

and that can't really happen anymore with this patch.

So yea lets make this public.